packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -1452,6 +1458,11 @@ void ArbCore::printCoreThreadBacktrace() {
 bool ArbCore::reorgIfInvalidMachine(uint32_t& thread_failure_count,
                                     uint256_t& next_checkpoint_gas,
                                     ValueCache& cache) {
+    if (core_machine->status() == MachineThread::MACHINE_RUNNING) {
+        // Just continue
+        return true;
+    }
```

Comment:
I'm not sure why we need this?

Reply:
Fixed to call `core_machine->finishThread()` instead.  `getReorgData` is not thread safe, so want to make certain that machine is done running.